File: go/btree/symmetric-binary-tree.go
package btree

import "fmt"

func isSymmetric(root *Node) bool {
	if root == nil {
		return true
	}
	return isMirror(root.Left, root.Right)
}

func isMirror(left *Node, right *Node) bool {
	if left == nil && right == nil {
		return true
	}
	if left != nil && right != nil && left.Val == right.Val {
		return true && isMirror(left.Left, right.Right) && isMirror(left.Right, right.Left)
	}
	return false
}

func RunIsSymmetric() {
	tree := Tree{
		Root: &Node{
			Val: 1,
			Left: &Node{
				Val: 2,
				Left: &Node{
					Val: 3,
				},
				Right: &Node{
					Val: 4,
				},
			},
			Right: &Node{
				Val: 2,
				Left: &Node{
					Val: 4,
				},
				Right: &Node{
					Val: 3,
				},
			},
		},
	}
	fmt.Println(isSymmetric(tree.Root))
}

File: php/symmetric-binary-tree.php
<?php

class Node {
    public $val;
    public $left;
    public $right;

    public function __construct($val) {
        $this->val = $val;
        $this->left = null;
        $this->right = null;
    }
}

class Tree {
    public $root;
}

class Solution {
    public static function isSymmetric(Node $root): bool {
        if(is_null($root)) {
            return true;
        }
        return self::isMirror($root->left, $root->right);
    }

    public static function isMirror(Node $left = null, Node $right = null): bool {
        if(is_null($left) && is_null($right)) {
            return true;
        }
        if(is_null($left) || is_null($right)) {
            return false;
        }
        return ($left->val == $right->val) && self::isMirror($left->left, $right->right) && self::isMirror($left->right, $right->left);
    }
}

$root = new Node(1);
$root->left = new Node(2);
$root->right = new Node(2);
$root->left->left = new Node(3);
$root->left->right = new Node(4);
$root->right->left = new Node(4);
$root->right->right = new Node(3);

var_dump(Solution::isSymmetric($root));